h2quic/client.go

```diff
@@ -44,57 +46,34 @@ var _ h2quicClient = &Client{}
 
 // NewClient creates a new client
 func NewClient(t *QuicRoundTripper, tlsConfig *tls.Config, hostname string) *Client {
-	c := &Client{
+	return &Client{
 		t:               t,
+		dialAddr:        quic.DialAddr,
 		hostname:        authorityAddr("https", hostname),
 		responses:       make(map[protocol.StreamID]chan *http.Response),
 		encryptionLevel: protocol.EncryptionUnencrypted,
+		config: &quic.Config{
+			TLSConfig:                     tlsConfig,
+			RequestConnectionIDTruncation: true,
+		},
+		handshakeChan: make(chan struct{}),
```

Comment:
I think the name collision with the handshakeChan in client.go is unfortunate, could you maybe rename this one here?